File: server/auth-service/src/main/java/com/example/messenger/security/service/UserService.java
package com.example.messenger.security.service;

import com.example.messenger.entity.UserEntity;
import com.example.messenger.models.User;
import com.example.messenger.payload.response.MessageResponse;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface UserService {
    ResponseEntity<MessageResponse> register(User user);
    void checkCredentials(String userId, String userSecret);
    UserEntity checkCredentialsByLogin(String login, String password);
    User getUser(Long id);
    List<User> getAllUsers();
}

File: client/messenger-app/src/services/user.service.js
import api from './api';

class UserService {
    getUserProfile(){
        return api.get("/user");
    }
    getUserFriends(){
        return api.get("/user/friends")
    }
    getUser(id){
        return api.get("/user/"+id)
    }
    
}
export default new UserService();